### ISSUE TITLE
Cloudpickling of class variable fails for subclass of typing.Generic

### ISSUE THREAD

**Issue by dannyfriar** [cloudpipe/cloudpickle#463]
Related to: https://github.com/cloudpipe/cloudpickle/issues/440

Run the following to create the object and cloudpickle to disk:
```
import typing
import cloudpickle

T = typing.TypeVar("T")

class Base(typing.Generic[T]):    
    foo = "bar"
    
class Child(Base):
    pass

child = Child()

with open("test.pkl", "wb") as f:
    cloudpickle.dump(child, f)
```

Then if load in another Python process:

```
import cloudpickle

with open("test.pkl", "rb") as f:
    obj = cloudpickle.load(f)
    
print(obj.foo)
```

Gives `AttributeError: 'Child' object has no attribute 'foo'`

Using `cloudpickle==2.0.0` and python version `3.8.12`. Observed on both Windows 10 and CentOS 8.4.

The above example works correctly (i.e. does not throw the `AttributeError`) with `cloudpickle==2.0.0` and python version `3.6.6`.

**Comment by dannyfriar** [comment]
Also fails if you set a property on the parent class:

```
class Base(typing.Generic[T]):    
    @property
    def foo(self):
        return "bar"
```

**Comment by pierreglaser** [comment]
Hi @dannyfriar, thanks for the report. We'll fix this in https://github.com/cloudpipe/cloudpickle/pull/448. Also, just as a FYI, is:
```python
import typing
import cloudpickle

T = typing.TypeVar("T")

class Base(typing.Generic[T]):    
    foo = "bar"
    
class Child(Base):
    pass
```
a legitimate use-case in typing? Regardless of the answer, we'll fix this issue in `cloudpickle`, but I'm just curious.

**Comment by dannyfriar** [comment]
Thanks @pierreglaser. I think this is a legitimate use case (obviously not the dummy example here). A better example from the mypy docs: https://mypy.readthedocs.io/en/stable/generics.html

**Comment by pierreglaser** [comment]
Closed by #448
